scripts/update-esbuild-versions.js

```diff
@@ -0,0 +1,106 @@
+    }
+  });
+
+  writeFileSync(filepath, fileContent);
+}
+
+function fetch(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      if (res.statusCode !== 200) {
+        console.error(res);
+        return reject();
+      }
+      
+      let body = '';
+      res.on("data", (chunk) => body += chunk);
+      res.on("end", () => resolve(JSON.parse(String(body))));
+    });
+  });
+}
+
+function downloadFile(url, dest) {
+  return new Promise((resolve, reject) => {
+    const file = createWriteStream(dest);
+
+    const request = https.get(url, (response) => {
+        response.pipe(file);
+    });
+
+    file.on('finish', () => {
+      file.end();
+      resolve();
+    });
+  });
+};
+
+async function main() {
+  const content = [];
+  const fileReplacements = [];
+
+  content.push('""" Generated code; do not edit\nUpdate by running yarn update-esbuild-versions\n\nHelper macro for fetching esbuild versions for internal tests and examples in rules_nodejs\n"""\n');
+  content.push('load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")\n')
+
+  const latestVersion = (await fetch('https://registry.npmjs.org/esbuild/latest')).version;
+  content.push(`_VERSION = "${latestVersion}"\n`);
+  fileReplacements.push([/_ESBUILD_VERSION = "(.+?)"/, latestVersion])
+
+  content.push('def esbuild_dependencies():');
+  content.push('    """Helper to install required dependencies for the esbuild rules"""\n');
+  content.push('    version = _VERSION\n');
+
+  const tmpDir = tmpdir();
+  mkdirSync(tmpDir, {recursive: true});
+
+  for(const platform of Object.keys(PLATFORMS)) {
+    const downloadUrl = `https://registry.npmjs.org/${PLATFORMS[platform]}/-/${PLATFORMS[platform]}-${latestVersion}.tgz`;
+
+    const downloadPath = join(tmpDir, PLATFORMS[platform]);
+    await downloadFile(downloadUrl, downloadPath);
+    const shasumOutput = exec(`shasum -a 256 ${downloadPath}`, { silent: true }).stdout;
+    const shasum = shasumOutput.split(' ')[0];
+
+    fileReplacements.push([new RegExp(`"${platform}",.+?sha256 = "(.+?)"`, 's'), shasum])
+
+    content.push('    http_archive(');
+    content.push(`        name = "${platform}",`);
+    content.push('        urls = [');
+    content.push(`            "https://registry.npmjs.org/${PLATFORMS[platform]}/-/${PLATFORMS[platform]}-%s.tgz" % version,`);
+    content.push('        ],');
+    content.push('        strip_prefix = "package",');
+    content.push(`        build_file_content = """exports_files(["${platform === 'esbuild_windows' ? 'esbuild.exe' : 'bin/esbuild'}"])""",`);
+    content.push(`        sha256 = "${shasum}",`);
+    content.push('    )');
+  }
+
+
+  rmdirSync(tmpDir, {recursive: true});
+
+  console.log(content.join('\n'));
+
+  // replace shasums in some manually edited files
+  replaceFileContent('examples/esbuild/WORKSPACE', fileReplacements);
+  replaceFileContent('packages/esbuild/_README.md', fileReplacements);
+}
+
+main();
```

Comment:
Does this always update to latest, or is there a way to force it to a version?

Reply:
Updated to allow for a version to be specified when running the script. Can be run as:
`node ./scripts/update-esbuild-versions.js 0.8.57`